runtime/caml/misc.h

```diff
@@ -293,14 +293,25 @@ CAMLextern void caml_failed_assert (char *, char_os *, int)
 #endif
 
 #if __has_builtin(__builtin_trap) || defined(__GNUC__)
-  #define CAMLunreachable() (__builtin_trap())
+  CAMLnoret Caml_inline void caml_abort(void) {
+    __builtin_trap();
+  }
 #elif defined(_MSC_VER)
   #include <intrin.h>
-  CAMLnoret Caml_inline void caml_fastfail(unsigned int);
-  void caml_fastfail(unsigned int i) { __fastfail(i); }
-  #define CAMLunreachable() (caml_fastfail(7 /* FAST_FAIL_FATAL_APP_EXIT */))
+  CAMLnoret Caml_inline void caml_abort(void) {
+    __fastfail(7 /* FAST_FAIL_FATAL_APP_EXIT */);
+  }
 #else
-  #define CAMLunreachable() (CAMLassert(0))
+  CAMLnoret Caml_inline void caml_abort(void) {
+    abort();
+  }
+#endif
+
+#ifdef DEBUG
+CAMLnoret CAMLextern void caml_debug_abort(char_os *, int); // in misc.c
```

Comment:
I think it is unnecessary to mention that a routine which prototype is in `misc.h` lives in `misc.c`.
```suggestion
CAMLnoret CAMLextern void caml_debug_abort(char_os *, int);
```